views/tx.tmpl

```diff
@@ -50,13 +70,24 @@
                     {{range .Vin}}
                     <tr>
                         <td>
-                            {{ if .Coinbase}}<span>{{ .Coinbase }}</span>{{end}}
-                            <span class="hash collapse">{{.Txid}}</span>
+                            {{if .Coinbase}}
+                                Coinbase: {{ .Coinbase }}
+                            {{else if eq .Txid "0000000000000000000000000000000000000000000000000000000000000000"}}
+                                Vote Reward
+                            {{else}}
+                            <a href="/explorer/tx/{{.Txid}}"><span class="hash collapse">{{.Txid}}</span></a>
+                            {{end}}
+                        </td>
+                        <td>
+                        {{if eq .BlockHeight 0}}
+                            N/A
+                        {{else}}
+                            <a href="/explorer/block/{{$.Block.BlockHash}}">{{.BlockHeight}}</a></td>
+                        {{end}}
                         </td>
-                        <td>{{if ne .BlockHeight 0}}{{.BlockHeight}}{{end}}</td>
                         <td>{{.Sequence}}</td>
                         <td>{{.Tree}}</td>
-                        <td>{{.AmountIn | getAmount}}</td>
+                        <td>{{if eq .AmountIn -1e-8}} N/A {{else}} {{.AmountIn}} {{end}}</td>
```

Comment:
Unless we understand clearly where -1e-8 is coming from (is it hard-coded somewhere?) I'd prefer if we change this to <0, but it's OK for now.